File: order.service/src/main/java/br/com/edu/order/application/service/OrderService.java
/* (C)2024 */
package br.com.edu.order.application.service;

import br.com.edu.order.domain.Order;
import br.com.edu.order.domain.repository.OrderRepository;
import java.util.List;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Slf4j
@Service
public class OrderService {

    private final OrderRepository repository;

    public OrderService(final OrderRepository repository) {
        this.repository = repository;
    }

    @Transactional
    public void save(final Order order) {
        var saved = repository.save(order);
        log.info("Saved order {}", saved.getId());
    }

    @Transactional
    public void saveAll(final List<Order> orders) {
        log.info("Inserting {} orders", orders.size());
        orders.forEach(Order::calculateTotal);
        orders.forEach(Order::changeProcessed);

        final var saved = repository.saveAll(orders);
        log.info("Saved {} orders", saved);
    }
}

File: order.service/order-service-usecase/src/main/kotlin/order/service/OrderService.kt
package order.service

import order.domain.Order
import order.repository.OrderRepository
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional
import java.util.function.Consumer

@Service
class OrderService (
    val repository: OrderRepository
) {

    private val logger = LoggerFactory.getLogger(OrderService::class.java)

    @Transactional
    fun save(order: Order) {
        val saved = repository.save(order)
        logger.info("Saved order {}", saved.id)
    }

    @Transactional
    fun saveAll(orders: List<Order>) {
        logger.info("Inserting {} orders", orders.size)
        orders.forEach(Consumer { obj: Order -> obj.calculateTotal() })
        orders.forEach(Consumer { obj: Order -> obj.changeProcessed() })

        val saved = mutableListOf<Order>()

        orders.forEach {
            saved.add(repository.save(it))
        }

        logger.info("Saved {} orders", saved)
    }

}
